presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -618,6 +619,11 @@ private boolean isGlacierObject(S3ObjectSummary object)
         return Glacier.toString().equals(object.getStorageClass());
     }
 
+    private static boolean isHadoopFolderMarker(S3ObjectSummary object)
+    {
+        return object.getKey().endsWith("_$folder$");
```

Comment:
Are you sure only the markers end with "_$folder$"? Should we check it is really an empty file?

Reply:
yeah these are files generated by Hadoop File system to indicate empty folders . More details are here - https://aws.amazon.com/premiumsupport/knowledge-center/emr-s3-empty-files/. There is already a reference of "_$folder$" in code https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java#L164 . This fix has already been applied on trinio - https://github.com/trinodb/trino/commit/d72622da2d04ec9ce56fe0a4e4772b548315cb2a. I have changed the PR commit message to include trinio commit id . I have also changed the code to use DIRECTORY_SUFFIX instead of "_$folders$" and added a check for size.

Reply:
Thanks. If this is cherry-picked from Trino, please add [co-authors](https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). Also, please sign CLA.